framework/architecture/spring/module-migration.rst

```diff
@@ -0,0 +1,239 @@
+Module migration
+================
+
+This guide explains how to migrate an existing Tocco module from HiveApp to Spring Boot.
+
+Create new module
+-----------------
+
+Apart from a few base modules, all modules should be created in either the ``core``, ``optional`` or ``customer``
+directories. Simply create a new directory named after the module (e.g. ``persist-core``) and create an empty
+``build.gradle`` file inside it.
+
+The new module needs to be registered in the ``settings.gradle`` file in the root directory of the project:
+
+  .. code-block:: text
+
+    include 'core:<module-name>'
+
+At this point it is recommended to refresh the Gradle project in IDEA in order to have IDE support for the next steps.
+
+We use the `Java Platform Module System <https://www.oracle.com/corporate/features/understanding-java-9-modules.html>`_
+for modularization. First we need to create the module descriptor ``module-info.java`` in the main source directory
+``src/main/java``:
+
+  .. code-block:: java
+
+    open module nice.core.reporting {
+
+    }
+
+The module name should be in the following format: ``nice.core/optional/customer.<module-name>``. Note that the module
```

Comment:
Would not be ``nice.{core, optional, customer}.<module-name>`` are more readable expression